src/aspire/classification/align2d.py

```diff
@@ -225,35 +376,620 @@ def align(self, classes, reflections, basis_coefficients):
+    ):
+        if not self.diagnostics:
+            return
+        labels = [0, 30, 60, 90, -60, -30]
+        x = y = np.arange(0, 180, 30) / (180 / cross_correlation.shape[0])
+        plt.title("Rotation Cross Correlation Map")
+        plt.imshow(cross_correlation)
+        plt.xlabel("Scale")
+        plt.yticks(y, labels, rotation="vertical")
+        plt.ylabel("Theta (Degrees)")
+        plt.show()
+
+        plt.plot(cross_correlation_score)
+        plt.title("Angle vs Cross Correlation Score")
+        plt.xticks(x, labels)
+        plt.xlabel("Theta (Degrees)")
+        plt.ylabel("Cross Correlation Score")
+        plt.grid()
+        plt.show()
+
+    def _rotated_diagnostic(self, ia, a, ib, b, sb, rb):
+        """
+        Plot the image after estimated rotation and reflection.
+
+        :param ia: index image `a`
+        :param a: image `a`
+        :param ib: index image `b`
+        :param b: image `b` after reflection `sb` and rotion `rb`
+        :param sb: Reflection, Boolean
+        :param rb: Estimated rotation, degrees
+        """
+
+        if not self.diagnostics:
+            return
+
+        fig, axes = plt.subplots(1, 2)
+        ax = axes.ravel()
+        ax[0].set_title(f"Image {ia}")
+        ax[0].imshow(a)
+        ax[0].grid()
+        ax[1].set_title(f"Image {ib} Refl: {str(sb)[0]} Rotated {rb:.1f}")
+        ax[1].imshow(b)
+        ax[1].grid()
+        plt.show()
+
+    def _translation_cross_corr_diagnostic(self, cross_correlation):
+        if not self.diagnostics:
+            return
+        plt.title("Translation Cross Correlation Map")
+        plt.imshow(cross_correlation)
+        plt.xlabel("x shift (pixels)")
+        plt.ylabel("y shift (pixels)")
+        L = self.alignment_src.L
+        labels = [0, 10, 20, 30, 0, -10, -20, -30]
+        tick_location = [0, 10, 20, 30, L, L - 10, L - 20, L - 30]
+        plt.xticks(tick_location, labels)
+        plt.yticks(tick_location, labels)
+        plt.show()
+
+    def _averaged_diagnostic(self, ia, a, ib, b, sb, rb):
+        """
+        Plot the stacked average image after
+        estimated rotation and reflections.
+
+        Compare in a three way plot.
+
+        :param ia: index image `a`
+        :param a: image `a`
+        :param ib: index image `b`
+        :param b: image `b` after reflection `sb` and rotion `rb`
+        :param sb: Reflection, Boolean
+        :param rb: Estimated rotation, degrees
+        """
+        if not self.diagnostics:
+            return
+        fig, axes = plt.subplots(1, 3)
+        ax = axes.ravel()
+        ax[0].set_title(f"{ia}")
+        ax[0].imshow(a)
+        ax[0].grid()
+        ax[1].set_title(f"{ib} Refl: {str(sb)[0]} Rot: {rb:.1f}")
+        ax[1].imshow(b)
+        ax[1].grid()
+        ax[2].set_title("Stacked Avg")
+        plt.imshow((a + b) / 2.0)
+        ax[2].grid()
+        plt.show()
+
+
+class BFSReddyChatterjiAlign2D(ReddyChatterjiAlign2D):
```

Comment:
I think the disc used in `BFSReddyChatterjiAlign2D` is much better than what was done in `BFRSAlign2D`... but TBH `BFRSAlign2D` should be deleted once anything better (perhaps the one in the PR) is the norm.  I can't imagine anyone using `BFSRAlign2D`...

Given choice, I'd just as soon delete the BFSRAlign2D... lmk.

Reply:
Right. That's what we said about `FBBasis` too though… It will be good to have a naive reference to compare the optimized versions (of which ReddyChatterji is one). I'll make an issue.